### PR TITLE
Replace raw streams with rmm::cuda_stream_view (part 3)

### PR DESCRIPTION
**Pull request by harrism**
Converting libcudf to use `rmm::cuda_stream_view` will require a LOT of changes, so I'm splitting it into multiple PRs to ease reviewing. This is the third PR in the series. This series of PRs will

- Replace usage of `cudaStream_t` with `rmm::cuda_stream_view`
- Replace usage of `0` or `nullptr` as a stream identifier with `rmm::cuda_stream_default`
- Ensure all APIs always order the stream parameter before the memory resource parameter. #5119

Contributes to #6645 and #5119

Depends on #6646 and #6648 so this PR will look much bigger until they are merged.

This third PR converts:

 - remaining dictionary functionality
 - cuio
 - lists
 - scalar
 - strings
 - groupby
 - join
 - contiguous_split
 - get_element
 - datetime_ops
 - extract
 - merge
 - partitioning
 - minmax reduction
 - scan
 - byte_cast
 - clamp
 - interleave_columns
 - is_sorted
 - groupby
 - rank
 - tests
 - concurrent map classes

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Review by mike-wendt — COMMENTED** [review]
Approved changes for our `CODEOWNERS` files; will rely on team to verify other changes

**Comment by harrism** [comment]
rerun tests

**Comment by kkraus14** [comment]
rerun tests

**Review by jlowe — APPROVED** [review]
Java approval.

File: cpp/src/strings/regex/regexec.cu [hunk 1]
```diff
@@ -18,6 +18,7 @@
 #include <cudf/detail/utilities/integer_utils.hpp>
 #include <rmm/device_buffer.hpp>
 #include <strings/regex/regex.cuh>
+#include "rmm/cuda_stream_view.hpp"
```
Comment (rgsl888prabhu) [inline cpp/src/strings/regex/regexec.cu:21]:
```suggestion
#include <rmm/cuda_stream_view.hpp>
```

File: cpp/src/strings/replace/backref_re.cu [hunk 2]
```diff
@@ -29,6 +29,7 @@
 #include <regex>
 
 #include "backref_re.cuh"
+#include "rmm/cuda_stream_view.hpp"
```
Comment (rgsl888prabhu) [inline cpp/src/strings/replace/backref_re.cu:32]:
```suggestion
#include <rmm/cuda_stream_view.hpp>
```

File: cpp/src/strings/replace/multi_re.cu [hunk 3]
```diff
@@ -26,6 +26,7 @@
 #include <strings/regex/regex.cuh>
 #include <strings/utilities.cuh>
 #include <strings/utilities.hpp>
+#include "rmm/cuda_stream_view.hpp"
```
Comment (rgsl888prabhu) [inline cpp/src/strings/replace/multi_re.cu:29]:
```suggestion
#include <rmm/cuda_stream_view.hpp>
```

File: cpp/src/strings/replace/replace_re.cu [hunk 4]
```diff
@@ -17,15 +17,16 @@
...
 #include <cudf/detail/nvtx/ranges.hpp>
-#include <cudf/null_mask.hpp>
 #include <cudf/strings/detail/utilities.hpp>
 #include <cudf/strings/replace_re.hpp>
 #include <cudf/strings/string_view.cuh>
 #include <cudf/strings/strings_column_view.hpp>
 #include <strings/regex/regex.cuh>
 #include <strings/utilities.cuh>
 #include <strings/utilities.hpp>
+#include "rmm/cuda_stream_view.hpp"
```
Comment (rgsl888prabhu) [inline cpp/src/strings/replace/replace_re.cu:29]:
```suggestion
#include <rmm/cuda_stream_view.hpp>
```

File: cpp/src/strings/substring.cu [hunk 5]
```diff
@@ -29,6 +29,7 @@
 #include <cudf/strings/substring.hpp>
 
 #include <strings/utilities.cuh>
+#include "rmm/cuda_stream_view.hpp"
```
Comment (rgsl888prabhu) [inline cpp/src/strings/substring.cu:32]:
```suggestion
#include <rmm/cuda_stream_view.hpp>
```

File: cpp/src/strings/translate.cu [hunk 6]
```diff
@@ -29,6 +29,7 @@
 #include <thrust/find.h>
 
 #include <algorithm>
+#include "rmm/cuda_stream_view.hpp"
```
Comment (rgsl888prabhu) [inline cpp/src/strings/translate.cu:32]:
```suggestion
#include <rmm/cuda_stream_view.hpp>
```

File: cpp/src/text/generate_ngrams.cu [hunk 7]
```diff
@@ -27,6 +27,7 @@
 #include <cudf/utilities/error.hpp>
 #include <nvtext/generate_ngrams.hpp>
 #include <strings/utilities.cuh>
+#include "rmm/cuda_stream_view.hpp"
```
Comment (rgsl888prabhu) [inline cpp/src/text/generate_ngrams.cu:30]:
```suggestion
#include <rmm/cuda_stream_view.hpp>
```

**Review by rgsl888prabhu — APPROVED** [review]
LGTM, just small changes

File: cpp/include/cudf/strings/detail/merge.cuh [hunk 8]
```diff
@@ -23,6 +23,7 @@
 #include <cudf/strings/string_view.cuh>
 #include <cudf/strings/strings_column_view.hpp>
 #include <strings/utilities.cuh>
+#include "rmm/cuda_stream_view.hpp"
```
[the remark this answers is not in the document]
Comment (harrism) [reply cpp/include/cudf/strings/detail/merge.cuh:26]: OMG I'm so sick of clangd sneaking these in. Thanks!

[the remark this answers is not in the document]
Comment (rgsl888prabhu) [reply cpp/include/cudf/strings/detail/merge.cuh]: Just reviewing drained me , I can't even imagine how you felt after three such PRs.

**Review by devavret — CHANGES_REQUESTED** [review]
Suggest making a debug build once and changing any `CHECK_CUDA(stream)` to `CHECK_CUDA(stream.view())`. There are several instances in this PR which I cannot request changes in because those lines are not part of diffs.

**Comment by harrism** [comment]
> Suggest making a debug build once and changing any `CHECK_CUDA(stream)` to `CHECK_CUDA(stream.view())`. There are several instances in this PR which I cannot request changes in because those lines are not part of diffs.

Thanks @devavret , done. Running a debug build in the background...

**Review by codereport — APPROVED** [review]
LGTM :+1:

**Comment by harrism** [comment]
@devavret I found and fixed an unrelated debug build issue. Can you approve?

**Review by devavret — APPROVED** [review]

**Pull request merged**